Darktable fails to build against SDL2 2.24.0

### Description

Darktable now fails to build against SDL2 2.24.0
MINGW64:
```
  [555/910] Linking C shared module lib\darktable\plugins\lighttable\libgamepad.dll
  FAILED: lib/darktable/plugins/lighttable/libgamepad.dll 
  cmd.exe /C "cd . && D:\M\msys64\mingw64\bin\gcc.exe -march=x86-64 -mtune=generic -O2 -pipe -Wall -Wno-format -Wshadow -Wtype-limits -Wvla -Wold-style-declaration -Wno-unknown-pragmas -Wno-error=varargs -Wno-format-truncation -Wno-error=address-of-packed-member -std=c99 -fopenmp -mtune=generic -msse2 -g -mfpmath=sse -O3 -DNDEBUG -O3 -ffast-math -fno-finite-math-only -fexpensive-optimizations  -pipe -Wl,--enable-runtime-pseudo-reloc   -Wl,--undefined=WinMain -shared -o lib\darktable\plugins\lighttable\libgamepad.dll -Wl,--major-image-version,0,--minor-image-version,0 @CMakeFiles\gamepad.rsp  && cmd.exe /C "cd /D C:\_\mingw-w64-darktable\src\build-MINGW64\lib\darktable\plugins\lighttable && D:\M\msys64\mingw64\bin\cmake.exe -E make_directory .debug && objcopy --only-keep-debug C:/_/mingw-w64-darktable/src/build-MINGW64/lib/darktable/plugins/lighttable/libgamepad.dll C:/_/mingw-w64-darktable/src/build-MINGW64/lib/darktable/plugins/lighttable/libgamepad.dll.dbg && objcopy --strip-debug C:/_/mingw-w64-darktable/src/build-MINGW64/lib/darktable/plugins/lighttable/libgamepad.dll && objcopy --add-gnu-debuglink=libgamepad.dll.dbg C:/_/mingw-w64-darktable/src/build-MINGW64/lib/darktable/plugins/lighttable/libgamepad.dll""
  D:/M/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/M/msys64/mingw64/lib/libSDL2main.a(SDL_windows_main.c.obj):(.text+0x152): undefined reference to `SDL_main'
```
CLANG64:
```
  [554/910] Linking C shared module lib\darktable\plugins\lighttable\libgamepad.dll
  FAILED: lib/darktable/plugins/lighttable/libgamepad.dll 
  cmd.exe /C "cd . && D:\M\msys64\clang64\bin\clang.exe -march=x86-64 -mtune=generic -O2 -pipe -fopenmp-version=45 -Wall -Wno-format -Wshadow -Wtype-limits -Wvla -Wthread-safety -Wno-unknown-pragmas -Wno-error=varargs -Wno-error=address-of-packed-member -std=c99 -fopenmp=libomp -mtune=generic -msse2 -g -O3 -DNDEBUG -O3 -ffast-math -fno-finite-math-only  -pipe -Wl,--enable-runtime-pseudo-reloc   -Wl,--undefined=WinMain -shared -o lib\darktable\plugins\lighttable\libgamepad.dll -Wl,--major-image-version,0,--minor-image-version,0 @CMakeFiles\gamepad.rsp  && cmd.exe /C "cd /D C:\_\mingw-w64-darktable\src\build-CLANG64\lib\darktable\plugins\lighttable && D:\M\msys64\clang64\bin\cmake.exe -E make_directory .debug && objcopy --only-keep-debug C:/_/mingw-w64-darktable/src/build-CLANG64/lib/darktable/plugins/lighttable/libgamepad.dll C:/_/mingw-w64-darktable/src/build-CLANG64/lib/darktable/plugins/lighttable/libgamepad.dll.dbg && objcopy --strip-debug C:/_/mingw-w64-darktable/src/build-CLANG64/lib/darktable/plugins/lighttable/libgamepad.dll && objcopy --add-gnu-debuglink=libgamepad.dll.dbg C:/_/mingw-w64-darktable/src/build-CLANG64/lib/darktable/plugins/lighttable/libgamepad.dll""
  ld.lld: error: undefined symbol: SDL_main
  >>> referenced by libSDL2main.a(SDL_windows_main.c.obj):(main_getcmdline)
```

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

Microsoft Windows Server 2022
 10.0.20348
 Datacenter

### MINGW environments affected

- [X] MINGW64
- [ ] MINGW32
- [X] UCRT64
- [X] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

### Expected behavior

Succeeds

### Actual behavior

Fails

### Repro steps

Rebuild darkatable against the last updates.

### Are you willing to submit a PR?

_No response_

--- a/mingw-w64-darktable/003-disable-gamepad.patch
+++ b/mingw-w64-darktable/003-disable-gamepad.patch
@@ -1,0 +1,20 @@
+--- a/src/libs/CMakeLists.txt
++++ b/src/libs/CMakeLists.txt
+@@ -97,7 +97,7 @@
+   add_library(midi MODULE "tools/midi.c")
+   target_link_libraries (midi ${PORTMIDI_LIBRARY})
+ endif()
+-
++#[[
+ find_package(SDL2)
+ if(SDL2_INCLUDE_DIRS)
+  add_definitions("-DHAVE_SDL")
+@@ -106,7 +106,7 @@
+   add_library(gamepad MODULE "tools/gamepad.c")
+   target_link_libraries(gamepad ${SDL2_LIBRARIES})
+ endif()
+-
++]]
+ if(BUILD_BATTERY_INDICATOR)
+   add_library(battery_indicator MODULE "tools/battery_indicator.c")
+ endif()

--- a/mingw-w64-darktable/PKGBUILD
+++ b/mingw-w64-darktable/PKGBUILD
@@ -4,7 +4,7 @@ _realname=darktable
 pkgbase=mingw-w64-${_realname}
 pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
 pkgver=4.0.0
-pkgrel=1
+pkgrel=2
 pkgdesc="darktable is an open source photography workflow application and raw developer (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw64' 'ucrt64' 'clang64' 'clangarm64')
@@ -46,11 +46,13 @@ depends=("${MINGW_PACKAGE_PREFIX}-dbus-glib"
 options=('!strip')
 source=("https://github.com/darktable-org/${_realname}/releases/download/release-${pkgver}/${_realname}-${pkgver}.tar.xz"{,.asc}
         "001-define-libraw-win32-unicodepaths-libcxx.patch"
-        "002-aarch64-mingw.patch")
+        "002-aarch64-mingw.patch"
+        "003-disable-gamepad.patch")
 sha256sums=('1416f8f59717e65a6220541aaa12eacca93888ce5176f2c9ab6c17b9cc53cc2d'
             'SKIP'
             'd586d8f56546eba5e70a4ac0ed99e43ce3d82685c48e77e9f40d0563965133e8'
-            'f350ecb20cab030180e5809f9fc7a0296a245f3461cf04f746c9bcbd87d12f86')
+            'f350ecb20cab030180e5809f9fc7a0296a245f3461cf04f746c9bcbd87d12f86'
+            '85f648d9d3c9bf39f17862cc120676456ee0a32c2c17f11c083057179647534a')
 validpgpkeys=('F10F9686652B0E949FCD94C318DCA123F949BD3B') # Pascal Obry <pascal@obry.net>
 
 apply_patch_with_msg() {
@@ -65,7 +67,8 @@ prepare() {
   cd "${srcdir}"/${_realname}-${pkgver}
   apply_patch_with_msg \
     001-define-libraw-win32-unicodepaths-libcxx.patch \
-    002-aarch64-mingw.patch
+    002-aarch64-mingw.patch \
+    003-disable-gamepad.patch
 }
 
 build() {

